dist/tools/doccheck/check.sh

```diff
@@ -74,3 +75,40 @@ then
     echo "${UNDEFINED_GROUPS_PRINT}"
     exit 2
 fi
+
+# Check for groups defined multiple times:
+# - remove empty spaces from the beginning of the line
+# - replace spaces with ':' for easily looping on the result when printing
+# - lines starting with '1:' are excluded
+MULTIPLE_DEFINED_GROUPS=$(echo "${DEFINED_GROUPS}" | uniq -c | \
+    sed 's/^ *//' | tr ' ' ':' | grep -v "^1:")
+
+# Remove the group description from each line containing a group definition.
+# Examples of input lines:
+# boards/nucleo-l152re/doc.txt:@defgroup boards_nucleo-l152re STM32 Nucleo-L152RE
+# boards/nucleo-l152re/include/periph_conf.h: * @defgroup boards_nucleo-l152re
+# Expected output lines patterns:
+# <filepath>: * <group name>
+# <filepath>:<group name>
+ALL_DEFGROUP_CLEANED=$(echo "${ALL_RAW_DEFGROUP}" | \
+    awk -F@ '{ split($2, end, " "); printf("%s%s\n",$1,end[2]) }')
+
+MULTIPLE_GROUPS_PRINT=$( \
+    for group in ${MULTIPLE_DEFINED_GROUPS}; \
+    do \
+        _group_name=$(echo ${group} | awk -F: '{ printf ("%s\n",$2) }'); \
+        _group_count=$(echo ${group} | awk -F: '{ printf ("%s\n",$1) }'); \
+        echo -e "\n${CWARN}${_group_name}${CRESET} defined ${CWARN}${_group_count}${CRESET} times in:"; \
```

Comment:
This is the only instance count is used and it could be replaced with "multiple", as you list them anyway below, so I think the simplicfication I proposed above can be applied.